docs/designs/template-000.md

```diff
@@ -0,0 +1,9 @@
+# Title
+
+## Overview
+
+## Context
+
+## [Free-form sections]
+
+## Decision
```

Comment:
What do you think about moving "Decision" up, right after "Overview"? If the doc is to be used as a quick reference it would be nice to have it closer to the problem statement. Also it looks like not every design requires a decision, "Implement as described here" seems to be implied by the type of the document, maybe it should be marked as [Optional]? Or maybe even made a part of free-form sections when there are several possible strategies? Overall it looks like "Context" will ultimately dictate the structure of everything that follows.

Reply:
> Also it looks like not every design requires a decision, "Implement as described here" seems to be implied by the type of the document, maybe it should be marked as [Optional]?

Agreed. I would probably go as far as omitting the `Decision` section completely for now. Remember, we're dealing with design docs for package managers, so the decision inherently is to "implement support for one". We can add it optional in a follow-up patch, we may get clearer picture and overall better feeling of this once we start actively using the template.

Reply:
We can mark Decision as optional. I agree it feels very out of place for initial package manage designs.

Reply:
IMO the "Decision" section here should be where we describe what we're going to implement given the "Context" described above. For a design, I would expect it to at a minimum include what is in-scope and out-of-scope.

We might just need a different word to describe it.